lib/net/ssh/multi/session.rb

```diff
@@ -441,6 +441,17 @@ def process(wait=nil, &block)
       end
     end
 
+    def io_select_wait(wait)
+      return wait if wait
```

Comment:
I based this on the implementation in net-ssh, where the wait argument takes precedence over the keepalive interval: https://github.com/net-ssh/net-ssh/blob/v3.0.2/lib/net/ssh/connection/session.rb#L611-L615

I agree though, using the smaller of the two values seems more correct to me. I'll fix it here, and submit a pull request to net-ssh to change the behaviour there too.